include/verilated_types.h

```diff
@@ -601,14 +615,20 @@ template <class T_Key, class T_Value> class VlAssocArray final {
     T_Value& at(const T_Key& index) {
         const auto it = m_map.find(index);
         if (it == m_map.end()) {
-            std::pair<typename Map::iterator, bool> pit = m_map.emplace(index, m_defaultValue);
-            return pit.first->second;
+            auto it2 = std::find_if(m_modify_input.begin(), m_modify_input.end(), [&](const auto& pair) {
+                return pair.first == index;
+            });
+            if (it2 == m_modify_input.end()) {
+                m_modify_input.emplace_back(index, m_defaultValue);
+                return m_modify_input.back().second;
+            }
+            return (*it2).second;
         }
         return it->second;
     }
```

Comment:
Roughly Test
1 thread  based test case

### Original

real    41m51.747s
user   41m47.964s
sys     0m3.516s

### Implementation at with map emplace
real    42m33.021s
user   42m29.767s
sys     0m2.916s

### m_modify_input and map()
real    2m25.726s
user    2m23.048s
sys     0m2.200s

### The observation
The emplace version is slower but it can be ignored due to experimental error, 
From the flamegraph observation, the bottleneck comes from the constructor of VlAssocArray, not at function

Reply:
Ok, so if the performance difference is due to the constructor (copy constructor in this case I presume), then what you really done here is replaced a `std::map<_, _>` with a `std::vector<std::pair<_, _>>`. Your performance gain is then due to copying a vector being faster, but you now have O(n) lookups instead of O(log(n)), which is bad. In practice, I presume `map()` is never actually called in your benchmark and `m_map.size()` is approximately zero. As soon as you have a single call to `map()`, and `m_map` is no longer small, you will see the performance regress to what it was. If that is the case, then the real fix should be to implement #3320.

Reply:
The code is our original design, but it fails on test_regression/t/t_assoc.v
due to 
``` cpp
template <std::size_t T_Words> struct VlWide final {
    bool operator<(const VlWide<T_Words>& rhs) const;
}
auto it2 = std::find_if(m_modify_input.begin(), m_modify_input.end(), [&](const auto& pair) {
    return pair.first == index;
});
```
the find_if cannot work always,  so we rewrite the code
``` cpp
    T_Value& at(const T_Key& index) {
        const auto it = m_map.find(index);
        if (it == m_map.end()) {
            if (m_modify_input.empty() ||
                (!m_modify_input.empty() && m_modify_input.back().first < index)) {
                m_modify_input.emplace_back(index, m_defaultValue);
                return m_modify_input.back().second;
            }
            const auto pair = std::make_pair(index, m_defaultValue);
            auto it2 = std::lower_bound(m_modify_input.begin(), m_modify_input.end(), pair,
                            [](const auto &value, const auto &input) {
                                   return !(input.first < value.first);
                            });
            if (index < it2->first)
                it2 = m_modify_input.insert(it2, std::make_pair(index, m_defaultValue));
            return it2->second;
        }
        return it->second;
    }
```
Now `m_modify_input` must be sorted, the search complexity is O(log(n)), not O(n) right now

Reply:
`m_modify_input.insert` is still O(n).

Reply:
Yes, but I think it is a reasonable tradeoff solution.
Maybe flat_map could enhance, but it doesn't belong to standard library

Reply:
I made another experiment base on this
The original method is 
``` cpp
            if (index < it2->first)
                it2 = m_modify_input.insert(it2, std::make_pair(index, m_defaultValue));
```
I change to this
``` cpp
            if (index < it2->first) {
                std::pair<typename Map::iterator, bool> pit = m_map.emplace(index, m_defaultValue);
                return pit.first->second;
            } else {
                return it2->second;
            }
            return it2->second;
```
It is guaranteed all operation is O(log(n))
But the test report

real    42m37.962s
user    42m34.804s
sys     0m2.928s

It's performance is downgrade to without any modification.